E.Deezer/Endpoint/BrowseEndpoint.cs

```diff
@@ -27,54 +21,48 @@ public interface IBrowseEndpoint
 
     internal class BrowseEndpoint : IBrowseEndpoint
     {
-        private readonly IGenreEndpoint iGenre;
-        private readonly IChartsEndpoint iCharts;
-        private readonly IUserEndpoint iUserEndpoint;
+        private readonly IUserEndpoint _userEndpoint;
 
-        private readonly DeezerClient iClient;
+        private readonly DeezerClient _client;
 
-        public BrowseEndpoint(DeezerClient aClient)
+        public BrowseEndpoint(DeezerClient client)
         {
-            iClient = aClient;
+            _client = client;
 
-            iGenre = new GenreEndpoint(iClient);
-            iCharts = new ChartsEndpoint(iClient);
-            iUserEndpoint = new UserEndpoint(aClient);
+            Genre = new GenreEndpoint(_client);
+            Charts = new ChartsEndpoint(_client);
+            _userEndpoint = new UserEndpoint(client);
         }
 
+        public IGenreEndpoint Genre { get; }
 
-        public IGenreEndpoint Genre => iGenre;
-
-        public IChartsEndpoint Charts => iCharts;
+        public IChartsEndpoint Charts { get; }
 
         public IUserEndpoint CurrentUser
         {
             get
             {
-                if(!iClient.IsAuthenticated)
+                if (!_client.IsAuthenticated)
                 {
                     throw new NotLoggedInException();
                 }
 
-                return iUserEndpoint;
+                return _userEndpoint;
             }
         }
 
-
         public async Task<IAlbum> GetAlbumById(uint albumId)
         {
             var p = new List<IRequestParameter>()
             {
                 RequestParameter.GetNewUrlSegmentParamter("id", albumId),
             };
 
-            // Will throw if Deezer error
-            var response = await iClient.GetDeezerObject<AlbumObjectResponse>("album/{id}", p)
+            var response = await _client.GetPlain<Album>("album/{id}", p)
```

Comment:
I don't belive this is correct. 

IIRC, GetPlai expects responses in the form:
```json
{
    "error" : null,
    "data": "...
}
```

Where as the API responses for getting a single item return many different forms of the json such as:

```json
{
    "albumName" : "...",
    "albumArtist": "...,
}
```
Or on error:
```json
{
    "error": 
    {
        "..." : "..."
    }
}
```

Or:
```json
{
    "status": 404,
    "msg": "Not Found"
}
```

And so should the user request an object with an id that doesn't exist this code will crash.

Reply:
If i remember correctly i didn't found any endpoint that responded in a form of
``` json
{
    "error" : null,
    "data": "...
}
```
Calling https://api.deezer.com/album/302127 (https://developers.deezer.com/api/explorer?url=album/302127) returns
``` json
{
  "id": "302127",
  "title": "Discovery",
  "upc": "724384960650",
  ...
}
```

If You request for an invalid album https://api.deezer.com/album/1 (https://developers.deezer.com/api/explorer?url=album/1)
then the server responsd:
``` json
{
  "error": {
    "type": "DataException",
    "message": "no data",
    "code": 800
  }
}
```
and to handle this scenario in the `Add server-side Exception checking to all (GET, POST, DELETE)` (6aee5b941f1eb6c052bc7388a0ab0b569af1fcc9) there is a new class [ExceptionFactory](https://github.com/projectgoav/E.Deezer/blob/6aee5b941f1eb6c052bc7388a0ab0b569af1fcc9/E.Deezer/ExceptionFactory.cs) to raise DeezerErrors when needed.